### PR TITLE
campaigns: use user email from GitHub for user matching

### LINKED ISSUE #17555: Extend GitHub token scope for campaigns by user:email
In order to match PR authors to Sourcegraph users, we want to query the email address of the actor. That, however requires us to extend the token scope by `user:email`. Also, we need some fallback handling, so in case the token has no permission to access that field, we don't ask for it and return just the author name.

### PR DESCRIPTION
**Pull request by LawnGnome**
This for real fixes #17555, and is the follow up to #18403.

The GraphQL query succeeds even without the `user:email` scope: in that case, an empty e-mail address is returned by GitHub, rather than an error. This will be handled the same as before.

### CODE REVIEW

File: internal/extsvc/github/testdata/golden/MarkPullRequestReadyForReview-1 [hunk 1]
```diff
@@ -2,21 +2,22 @@
   "ID": "MDExOlB1bGxSZXF1ZXN0NTA2MDg0ODU2",
   "Title": "This is a test PR, feel free to ignore",
   "Body": "I'm opening this PR to test something. Please ignore.",
-  "State": "OPEN",
+  "State": "CLOSED",
```
Comment (eseliger) [inline internal/extsvc/github/testdata/golden/MarkPullRequestReadyForReview-1:5]: does this request work when it's closed? 🤔

**Review by eseliger — APPROVED** [review]
nice! what a diff 😬

File: internal/extsvc/github/pulls.go [hunk 2]
```diff
@@ -1279,6 +1285,9 @@ fragment pr on PullRequest {
   %s
   author {
     ...actor
+    ... on User {
```
Comment (mrnugget) [inline internal/extsvc/github/pulls.go:1288]: So, just for my understanding, this is the secret sauce here? `... on User` will only return something when the right scope is used?

**Review by mrnugget — APPROVED** [review]

**Comment by chrispine** [comment]
What's the status of this PR?

**Comment by LawnGnome** [comment]
> What's the status of this PR?

I'll close it out: we might pick it back up as the base of a new PR if and when #17555 comes back, but it's irrelevant right now, as the list of conflicts would indicate.

**Pull request closed without merging**